test/plugin_helper/test_server.rb

```diff
@@ -381,6 +381,24 @@ class Dummy < Fluent::Plugin::TestBase
         d2.stop; d2.before_shutdown; d2.shutdown; d2.after_shutdown; d2.close; d2.terminate
       end
     end
+
+    test 'close all connections by shutdown' do
+      @d.server_create_tcp(:s, @port) do |data, conn|
+      end
+
+      client_sockets = []
+      5.times do
+        client_sockets << TCPSocket.open("127.0.0.1", @port)
+      end
+      waiting(4){ sleep 0.1 until @d.instance_variable_get(:@_server_connections).size == 5 }
+
+      @d.before_shutdown
+      @d.shutdown
```

Comment:
```suggestion
      @d.stop
      @d.before_shutdown
      @d.shutdown
```

It would be better to run `stop` to follow the actual behavior.